src/graphics/webgl-layer.js

```diff
@@ -379,7 +377,7 @@ Crafty.webglLayerObject = {
         visible_gl.length = 0;
         for (i=0; i < l; i++) {
             current = q[i];
-            if (current._visible && current.__c.WebGL && current.program) {
+            if (current._visible && current.__c.WebGL && current.program && (current._drawLayer === this)) {
```

Comment:
Same answer, that our search for visible entities returns *all* entities.  (Although it looks like there's still a now redundant check for the WebGL component here.)